### PR TITLE
Make the TentacleClient decorators async and add a proxy so that sync calls can use the async decorators

### PR DESCRIPTION
**Pull request by nathanwoctopusdeploy**
# Background

Make the TentacleClient decorators async and add a proxy so that sync calls can use the async decorators.

This is in preparation for making the tests support both sync and async service calls

# How to review this PR

Quality :heavy_check_mark:

# Pre-requisites

- [ ] I have read [How we use GitHub Issues](https://github.com/OctopusDeploy/Issues/blob/master/docs/CONTRIBUTING.internal.md) for help deciding when and where it's appropriate to make an issue.
- [ ] I have considered informing or consulting the right people, according to the [ownership map](https://whimsical.com/ownership-map-NzbiD4HJyvhC9jNJNfS6TG).
- [ ] I have considered appropriate testing for my change.

### CODE REVIEW

File: source/Octopus.Tentacle.Tests.Integration/ClientScriptExecutionCanBeCancelledWhenRetriesAreEnabled.cs [hunk 1]
```diff
@@ -402,11 +427,16 @@ public async Task DuringGetStatus_ScriptExecutionCanBeCancelled(TentacleType ten
         }
 
         [Test]
-        [TestCase(TentacleType.Polling, RpcCallStage.Connecting)]
-        [TestCase(TentacleType.Listening, RpcCallStage.Connecting)]
-        [TestCase(TentacleType.Polling, RpcCallStage.InFlight)]
-        [TestCase(TentacleType.Listening, RpcCallStage.InFlight)]
-        public async Task DuringCompleteScript_ScriptExecutionCanBeCancelled(TentacleType tentacleType, RpcCallStage rpcCallStage)
+        [TestCase(TentacleType.Polling, RpcCallStage.Connecting, SyncOrAsyncHalibut.Sync)]
```
Comment (nathanwoctopusdeploy) [inline source/Octopus.Tentacle.Tests.Integration/ClientScriptExecutionCanBeCancelledWhenRetriesAreEnabled.cs:430]: This shouldn't be included in this PR

**Review by sburmanoctopus — APPROVED** [review]
